### PR TITLE
Removed BundleLookupTag from the ByoClusterSpec

### LINKED ISSUE #567: Remove BundleLookupTag from the ByoClusterSpec
Initially `BundleLookupTag` field was introduced in the [ByoClusterSpec](https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/blob/main/apis/infrastructure/v1beta1/byocluster_types.go#L30) as the BYOH Bundle OCI image tagging uses the `BYOH` release versioning scheme. Now the Bundle tags use the K8s version in the OCI image tag for e.g. `byoh-bundle-ubuntu_20.04.1_x86-64_k8s:v1.23.5`, hence we can remove the field and refactor the code to use K8s version.

### PR DESCRIPTION
**Pull request by Ankitasw**
**What this PR does / why we need it**:
This PR removes `BundleLookupTag` field from ByoClusterSpec and makes use of only k8s version in the BYOH OCI image tag. It also removes `byocluster_webhook` as no validations are required anymore.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #567

### CODE REVIEW

File: apis/infrastructure/v1beta1/byocluster_webhook.go
Comment (sachinkumarsingh092) [inline apis/infrastructure/v1beta1/byocluster_webhook.go:63]: This webhook is pretty much non-functional and empty. Do we even need this anymore?

Comment (sachinkumarsingh092) [inline apis/infrastructure/v1beta1/byocluster_webhook.go:63]: This webhook is pretty much non-functional and empty. Do we even need this anymore?

Comment (Ankitasw) [reply apis/infrastructure/v1beta1/byocluster_webhook.go:63]: It is affecting a lot of code which required all the more refactoring, that's why didn't touch it, but it is doable.

Comment (sachinkumarsingh092) [reply apis/infrastructure/v1beta1/byocluster_webhook.go:63]: Seems like we'll be removing `byoclustertemplate_webhook` as part of #573. We should similarly completely remove the `byocluster_webhook` here. WDYT?

Comment (Ankitasw) [reply apis/infrastructure/v1beta1/byocluster_webhook.go]: That makes sense to me, removed `byocluster_webhook`. Thanks for the suggestion.

**Review by sachinkumarsingh092 — APPROVED** [review]
LGTM

File: agent/host_agent_suite_test.go
Comment (anusha94) [inline agent/host_agent_suite_test.go:53]: @dharmjit Are we still at this version? Oh so only e2e we updated to v1.23.5 and integration tests use v1.22.3?

File: apis/infrastructure/v1beta1/byoclustertemplate_webhook.go
Comment (anusha94) [inline apis/infrastructure/v1beta1/byoclustertemplate_webhook.go:4]: possible conflict with PR #573 ?
I am referring to this comment - https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/pull/573#discussion_r885609665

**Review by anusha94 — COMMENTED** [review]
lgtm otherwise

File: apis/infrastructure/v1beta1/byoclustertemplate_webhook.go
Comment (Ankitasw) [reply apis/infrastructure/v1beta1/byoclustertemplate_webhook.go:4]: Yes I noticed it, but since we removed the `byocluster_webhook`, hence this change.

Comment (Ankitasw) [reply apis/infrastructure/v1beta1/byoclustertemplate_webhook.go:4]: What we can do is wait for #573 to merge maybe, before merging this one, wdyt @anusha94 ?

Comment (anusha94) [reply apis/infrastructure/v1beta1/byoclustertemplate_webhook.go:4]: I think we can merge this here and handle conflicts in that one. Basically, if the webhook is not doing anything, we should remove the file itself.

**Review by anusha94 — APPROVED** [review]

File: agent/host_agent_suite_test.go
Comment (dharmjit) [reply agent/host_agent_suite_test.go:53]: We could update `k8s` version for agent tests as well.

**Pull request merged**
